### PR TITLE
feat: Change a key for the dependency cache - WPB-14445

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14445" title="WPB-14445" target="_blank">BugWPB-14445</a>  [iOS]  Change a key for dependencies cache
  </td></table>
  <br />
 

### Issue

We create a key for the cache when building the app as: `hash(Package.resolved)-ENABLE_DATADOG`
But for tests we can hardcode the value of `ENABLE_DATADOG`, and this will allow the cache to be reused.

### Testing

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: .github/workflows/_reusable_run_tests.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: ${{ env.PACKAGES_DIR }}
-          key: ${{ runner.os }}-swiftpm-project-${{ hashFiles('**/*.xcworkspace/xcshareddata/swiftpm/Package.resolved') }}
+          key: ${{ runner.os }}-swiftpm-project-${{ hashFiles('**/*.xcworkspace/xcshareddata/swiftpm/Package.resolved') }}-true
```
Comment (caldrian) [inline .github/workflows/_reusable_run_tests.yml]: Looks good, however, I'm not sure if it should be `false` instead.

**Review by caldrian — APPROVED** [review]

File: .github/workflows/_reusable_run_tests.yml (hunk 1 above)
Comment (KaterinaWire) [reply .github/workflows/_reusable_run_tests.yml]: I thought the default `ENABLE_DATADOG `value is true, but I think it shouldn't be a problem to use cache with DD.

Comment (netbe) [reply .github/workflows/_reusable_run_tests.yml]: for running tests I would say false also. ENABLE_DATADOG is not defined here

**Review by netbe — CHANGES_REQUESTED** [review]
let's change it to false as it makes more sense, no DD for tests at all

**Review by netbe — APPROVED** [review]
looks good

**Pull request merged**